third_party/libuv_test.patch

```diff
@@ -0,0 +1,56 @@
+diff --git a/test/test-timer.c b/test/test-timer.c
```

Comment:
This looks like a general change to the libuv test code, nothing gRPC-specific. I don't see why we'd want to maintain patches for general improvements that they'd likely accept and maintain upstream. Patches are a maintenance burden best avoided, if possible. For example, if it does not apply cleanly onto the next libuv version, upgrading the libuv dependency now means we'd have to port a patch before we could do the upgrade.

Reply:
Thanks. Please collect the patches separately, and remove them from this PR. If it was a patch for something that we absolutely need for libuv to _function_ on iOS, the extra burden is justifiable. But this is not essential, and again, every patch we have to maintain adds a real maintenance burden for library upgrades.

Reply:
Chatted w/ Hannah offline, we have a solution to run iOS test without patching libuv via [ios_ui_test](https://github.com/bazelbuild/rules_apple/blob/master/doc/rules-ios.md#ios_ui_test). This allows us to re-launch the app per test case via [XCUIApplication.launch](https://developer.apple.com/documentation/xctest/xcuiapplication/1500467-launch?language=objc) if needed in order to simulate the same fork run behavior as other platform.

Reply:
unfortunately, re-launch ios ui app didn't work, it might because that it only restarts the ios ui toolkit part, not the entire process.
PS: ios ui test also runs super slow due to boot up the simulator.


If we cannot add patch in our build system, could we try upstream it to libuv and wait for them to merge first?

Another alternative maybe split LibuvTest​ target into smaller ones, though it'll make the project a bit messy.



Error log with ios ui test:

Assertion failed in external/com_github_libuv_libuv/test/test-timer.c on line 152: once_cb_called == 1
...
Failing tests:
Runner:
-[LibuvTimerTests testTimerStartTwice]

** TEST EXECUTE FAILED **

Reply:
discussed offline, we will add back timer_start_twice test in a follow up PR pending the process relaunch fix